perffile/events.go

```diff
@@ -37,7 +37,10 @@ type EventGeneric struct {
 func (g *EventGeneric) Decode() Event {
 	switch g.Type {
 	case EventTypeHardware:
-		return EventHardware(g.ID)
+		return EventHardware{
+			ID:        EventHardwareID(g.ID),
```

Comment:
Mask `g.ID & 0xFF` before casting.

Reply:
I think the uint8 is fine. I agree that it would probably be better to have explicit masking everywhere, but since we don't have it anywhere right now, adding that would be a separate change.